src/main/scala/sigmastate/serialization/trees/Relation2Serializer.scala

```diff
@@ -9,7 +9,7 @@ import sigmastate.serialization.{Serializer, ValueSerializer}
 
 case class Relation2Serializer[S1 <: SType, S2 <: SType, R <: Relation[S1, S2]]
 (override val opCode: Byte,
```

Comment:
@greenhat why not `R <: Value[SBoolean.type]`?

Reply:
@ergomorphic need to cast to `Relation` in `serializeBody` stopped me earlier. Let me try to go down that path again. I get that narrow type does not matter much here since in `ValueSerializer` it's all `Value[SType]` anyway.